### PR TITLE
Fix: marker not showing up

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Comment by ClobberXD** [comment]
Ah.. I must've accidentally removed this in #10. Thanks.

- No need for the `if` check.
- Needs some code-style changes.

```lua
minetest.register_on_leaveplayer(function(player)
    marker[player:get_player_name()] = nil
end)
```

File: mods/marker/init.lua [hunk 1]
```diff
@@ -55,3 +55,12 @@ minetest.register_chatcommand("marker",
...
 								minetest.registered_chatcommands["clrmrkr"])
+
+minetest.register_on_leaveplayer(
+	function(player)
+		local name = player:get_player_name()
+		if marker[name] then
+			marker[name] = nil
+		end
+	end
+)
```
Comment (ClobberXD) [inline mods/marker/init.lua]: The symbol above indicates that there's no new line at the end. Please add one.

**Comment by LoneWolfHT** [comment]
Why not add a check in the marker cmd that sets it to nil if it already exists

**Comment by ghost** [comment]
Okay.

**Comment by ClobberXD** [comment]
Nice. Could you please also do what Lone_WolfHT suggested? That is, removing an existing marker when calling `/mrkr` or `/marker`, by adding a simple `if` check in the chatcommand def above L21 like so:

```lua
if marker[name] then
    player:hud_remove(marker[name])
    marker[name] = nil
end
```

Also, the indentation should be converted to tabs instead of spaces within the `on_leaveplayer` callback.

**Comment by ghost** [comment]
Okay but whats wrong with updating the current hud?.

**Comment by ghost** [comment]
There I removed the spaces and added the thing lone_wolf wanted.

**Comment by LoneWolfHT** [comment]
I'm not good at reading the changes page but from what I've seen that's not what I was saying.
The reason the marker isn't showing up is because when a player leaves the game the markers table still has a HUD id in it. So when the player runs `/marker` when the code gets to that if-else statement in the `/marker` cmd it goes to the if statement and tries to change a HUD that doesn't exist. Which results in no way-point being set
My idea was to do something line this:
```lua
func = function(name, param)
	local x, y, z = string.match(param, "^([%d.-]+)[, ] *([%d.-]+)[, ] *([%d.-]+)$")
	local player = minetest.get_player_by_name(name)

	if not x or not y or not z then
		return false, "You must provide 3 coordinates!"
	end

	if marker[name] then
		marker[name] = nil
	end

	marker[name] = player:hud_add({
		hud_elem_type = "waypoint",
		name = x .. ", " .. y .. ", " .. z,
		number = 0xFF0000,
		world_pos = {x = x, y = y, z = z}
	})

	return true
end
```

That would mean you don't need the `register_on_leaveplayer()` function and you also remove quite a bit of code.
Don't use that code though. There's an issue where a player will start multiple markers at once

**Comment by ghost** [comment]
Excess me but when you see my code in the repo it looks like this:
```lua
local marker = {}

minetest.register_chatcommand("mrkr", {
	params = "<x> <y> <z>",
	description = "Adds a waypoint marker at the selected position.",
	privs = {interact = true},
	func = function(name, param)
		local x, y, z = string.match(param, "^([%d.-]+)[, ] *([%d.-]+)[, ] *([%d.-]+)$")
		local player = minetest.get_player_by_name(name)

		if (not x or not y or not z) and param ~= "" then
			return false, "You must provide 3 coordinates!"
		elseif param == "" then
			local pos = vector.round(player:get_pos())

			x = pos.x
			y = pos.y
			z = pos.z
		end
		
		if marker[name] then
			player:hud_remove(marker[name])
			marker[name] = nil
		end

		marker[name] = player:hud_add({
			hud_elem_type = "waypoint",
			name = x .. ", " .. y .. ", " .. z,
			number = 0xFF0000,
			world_pos = {x = x, y = y, z = z}
		})

		minetest.chat_send_player(name, "Marker set to: "..x..", "..y..", "..z)
		return true
	end
})

minetest.register_chatcommand("clrmrkr", {
	params = "",
	description = "Removes the marker waypoint.",
	privs = {},
	func = function(name)
		local player = minetest.get_player_by_name(name)

		if player and marker[name] then
			player:hud_remove(marker[name])
			marker[name] = nil
			minetest.chat_send_player(name, "Marker removed")
		end
		return true
	end
})

minetest.register_chatcommand("marker",
								minetest.registered_chatcommands["mrkr"])
minetest.register_chatcommand("clearmarker",
								minetest.registered_chatcommands["clrmrkr"])

minetest.register_on_leaveplayer(function(player)
	marker[player:get_player_name()] = nil
end)

```
And having **register_on_leaveplayer()** is a good idea because it unloads memory.

**Comment by LoneWolfHT** [comment]
Ok I have no problems with this then lol.
I need to learn how to read those changes lists...

File: mods/marker/init.lua [hunk 2]
```diff
@@ -17,19 +17,19 @@ minetest.register_chatcommand("mrkr", {
 			y = pos.y
 			z = pos.z
 		end
-
+
```
Comment (ClobberXD) [inline mods/marker/init.lua]: Accidental trailing whitespace

**Comment by ClobberXD** [comment]
@LoneWolfHT The `on_leaveplayer` callback is still required, because `marker[name]` will be set to `nil` only when the player uses `/mrkr`. If it isn't cleared when the player leaves, we're stuck with the exact problem this PR was trying to fix :)

**Comment by LoneWolfHT** [comment]
Doesn't the player have to use /marker to realize it isn't showing?

**Comment by ClobberXD** [comment]
Take this situation for example:

- A player `abc` logs in for the first time.
- Sets a marker.
- Leaves. `marker["abc"]` hasn't been cleared.

We can't expect them to join again, for `marker["abc"]` to get cleared/updated. What'd happen if they don't join again? It's good practice to free up memory by de-initializing objects and vars once they're no longer needed.

File: mods/marker/init.lua (hunk 2 above)
Comment (ClobberXD) [reply mods/marker/init.lua]: @Coder12a The latest commit hasn't fixed the trailing whitespace at L20

**Review by ClobberXD — APPROVED** [review]
LGTM

**Pull request closed without merging**